lib/bundler/inject.rb

```diff
@@ -1,6 +1,42 @@
 require "bundler/inject/version"
 require "bundler/inject/dsl_patch"
 
+module Bundler
+  module Inject
+    class << self
+      # Check if we should skip outputing warnings
+      #
+      # This can be set in two ways:
+      #
+      #   - Via bundler's Bundler::Settings
```

Comment:
minor, but should this be `via Bundler.settings`?  (or that ::Settings expected?)

Reply:
I guess this depends on how you look at it.

`Bundler.settings`, as you might know, is just a singleton interface for a `Bundler::Settings` instance:

https://github.com/bundler/bundler/blob/264ef6be/lib/bundler.rb#L272-L276

Similar to `Bundler.ui` and `Bundler.definition`.  So what I saying here is that the setting can be set via the `Bundler::Settings` class, not the `Bundler.settings` interface.  Either is "correct", I just chose to reference the class here since it actually contains all the logic... which leads to your other question...